src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -150,60 +149,64 @@ private Properties createConfig(String nodeAddr) {
     producerConfig.put(ProducerConfig.SEND_BUFFER_CONFIG, this.sndBuf);
     producerConfig.put(ProducerConfig.LINGER_MS_CONFIG, this.linger);
     producerConfig.put(ProducerConfig.RECEIVE_BUFFER_CONFIG, this.rcvBuf);
-    producerConfig.put(
-      ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
-      "org.apache.kafka.common.serialization.StringSerializer");
-    producerConfig.put(
-      ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
-      "com.emc.mongoose.storage.driver.kafka.io.DataItemSerializer");
+    try {
+      producerConfig.put(
+          ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+          Class.forName("org.apache.kafka.common.serialization.StringSerializer"));
+      producerConfig.put(
+          ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+          Class.forName("com.emc.mongoose.storage.driver.kafka.io.DataItemSerializer"));
+    } catch (ClassNotFoundException e) {
+      LogUtil.exception(Level.DEBUG, e, "{}: operation failed", stepId);
+    }
     return producerConfig;
   }
 
   private void submitRecordCreateOperation(final DataOperation recordOp, String nodeAddr) {
     try {
       val config = configCache.computeIfAbsent(nodeAddr, this::createConfig);
       val adminConfig =
-        adminClientCreateFuncCache.computeIfAbsent(config, AdminClientCreateFunctionImpl::new);
+          adminClientCreateFuncCache.computeIfAbsent(config, AdminClientCreateFunctionImpl::new);
       val adminClient = adminClientCache.computeIfAbsent(nodeAddr, adminConfig);
       val producerConfig =
-        producerCreateFuncCache.computeIfAbsent(config, ProducerCreateFunctionImpl::new);
+          producerCreateFuncCache.computeIfAbsent(config, ProducerCreateFunctionImpl::new);
       val kafkaProducer = producerCache.computeIfAbsent(nodeAddr, producerConfig);
```

Comment:
Is the `kafkaProducer` thread safe? The cached pool should be used if not. See the example in the pravega storage driver @ master branch.

Reply:
kafkaProducer is thread safe
however, probably we still should use the pool with the size = `ioWorkerCount` for the multithreaded producing which may be more efficient...